sdk/core/core-paging/src/models.ts

```diff
@@ -31,3 +31,22 @@ export interface PagedAsyncIterableIterator<T, PageT = T[], PageSettingsT = Page
    */
   byPage: (settings?: PageSettingsT) => AsyncIterableIterator<PageT>;
 }
+
+/**
+ * An interface that describes how to communicate with the service.
+ */
+export interface PagedResult<TPage, TPageSettings = PageSettings> {
+  /**
+   * Link to the first page of results.
+   */
+  link: string;
+  /**
+   * A method that returns a page of results.
+   */
+  getPage: (link: string, maxPageSize?: number) => Promise<{ page: TPage; nextLink?: string }>;
+  /**
+   * a function to implement the `byPage` method on the paged async iterator. The default is
+   * one that sets the `maxPageSizeParam` from `settings.maxPageSize`.
+   */
+  byPage?: (settings?: TPageSettings | undefined) => AsyncIterableIterator<TPage>;
```

Comment:
why is settings both optional and potentially undefined? I feel like it should be one or the other

Reply:
That slipped for some reason, fixed.